soaktest/soaktest.go

```diff
@@ -0,0 +1,240 @@
+			f, ok := partition.(float64)
+			if !ok {
+				fmt.Printf("Failed to convert partition"+
+					"to float64 %v\n", ok)
+				return ok
+			}
+			if len(parts) > 0 {
+				parts += ","
+			}
+			parts += fmt.Sprintf("%d", int(f))
+		}
+
+		rttValue := broker.(map[string]interface{})["rtt"]
+		avg, ok := rttValue.(map[string]interface{})["avg"].(float64)
+		if !ok {
+			fmt.Printf("Failed to convert avg to float64 %v\n", ok)
+			return ok
+		}
+
+		p99, ok = rttValue.(map[string]interface{})["p99"].(float64)
+		if !ok {
+			fmt.Printf("Failed to convert p99 to float64 %v\n", ok)
+			return ok
+		}
+
+		nid, ok := broker.(map[string]interface{})["nodeid"].(float64)
+		if !ok {
+			fmt.Printf("Failed to convert nodeid to float64 %v\n", ok)
+			return ok
+		}
+		nodeid = int(nid)
+
+		tags := []string{fmt.Sprintf("broker:%d, type: %s",
+			nodeid, raw["type"])}
+		DatadogGauge(BROKER_RTT_P99+topic, p99/1000000.0, tags)
+		DatadogGauge(BROKER_RTT_AVG+topic, avg/1000000.0, tags)
+	}
+	return true
+}
+
+// HandleStatsEvent converts Stats events to hashmap
+func HandleStatsEvent(e *kafka.Stats, topic string) error {
+	var raw map[string]interface{}
+	err := json.Unmarshal([]byte(e.String()), &raw)
+	if err != nil {
+		fmt.Printf("json unmarshall error: %s", err)
+		return err
+	}
+
+	ok := rttStats(raw, topic)
+	if !ok {
+		return fmt.Errorf("Failed to calculate broker rtt statistics")
+	}
+	return nil
+}
+
+// ProducerDeliveryCheck handles delivery report for producer
+func ProducerDeliveryCheck(e *kafka.Message) {
+	if e.TopicPartition.Error != nil {
+		DrErrCnt += 1
+		DatadogIncrement(PRODUCER_DR_ERR_TO+
+			*e.TopicPartition.Topic, 1)
+		fmt.Printf("Delivery failed: %v\n", e.TopicPartition)
+	} else {
+		DrCnt += 1
+		DatadogIncrement(PRODUCER_DR_TO+
+			*e.TopicPartition.Topic, 1)
+		fmt.Printf("Delivered message to topic at events channel "+
+			"%s [%d] at offset %d\n", *e.TopicPartition.Topic,
+			e.TopicPartition.Partition, e.TopicPartition.Offset)
+	}
+}
+
+func PrintConsumerStatus() {
+	fmt.Printf("consumer: %d messages consumed, %d duplicates, "+
+		"%d missed, %d message errors, %d consumer client-level errors\n",
+		ConsumerMsgCnt, MsgDupCntForOutput, MsgMissCntForOutput,
+		ConsumerMsgErrs, ConsumerErr)
+}
+
+func PrintProducerStatus() {
+	fmt.Printf("producer: %d messages produced, %d delivered, %d failed "+
+		"to deliver, %d producer client-level errors\n",
+		ProducerMsgId, DrCnt, DrErrCnt, ProducerErr)
+}
+
+// HandleMessage handles received messages, monitors latency and
+// verifies messages
+func HandleMessage(e *kafka.Message,
+	hwmarks map[string]uint64) bool {
+	if e.TopicPartition.Error != nil {
+		ConsumerMsgErrs += 1
+		DatadogIncrement(CONSUMER_RECEIVE_ERROR_FROM+
+			*e.TopicPartition.Topic, 1)
+		fmt.Printf("Message receive failed from: %s\n",
+			e.TopicPartition.Error)
+		return false
+	}
+	DatadogIncrement(CONSUMER_RECEIVE_FROM+*e.TopicPartition.Topic, 1)
+	fmt.Printf("consumer out Message on %s:\n%s\n",
```

Comment:
We probably don't want to print each message or the logs will fill up real quick